### PR TITLE
Added Upload feature / Added more SQL Insert and update method to mak…

### PR DESCRIPTION
**Pull request by PradeepKumar100791**
…e easy requests

### CODE REVIEW

File: lib/util.js [hunk 1]
```diff
@@ -209,5 +219,95 @@ export default {
...
+            Object.keys(data).forEach(key => {
+                this.buildFormData(formData, data[key], parentKey ? `${parentKey}[${key}]` : key);
+            });
+        } else {
+            const value = data == null ? '' : data;
+            formData.append(parentKey, value);
+        }
+    },
+
+    jsonToFormData: function (data) {
```
Comment (durlabhjain) [inline lib/util.js:308]: We can use have buildFormData or jsonToFormData with signature:

buildFormData: function(data, formData = new FormData(), parentKey)

**Review by durlabhjain — CHANGES_REQUESTED** [review]

File: lib/index.js [hunk 2]
```diff
@@ -94,17 +94,17 @@ class Framework {
...
+    async query({ controller, params, body, method = "POST" }) {
         const { serverUrl, client } = this;
         const url = `${serverUrl}/Controllers/${controller}.ashx`;
         let result;
         if (method === "POST") {
             if (typeof params?.toFormData === 'function') {
                 params = params.toFormData();
             }
-            result = await client.post(url, { form: params });
+            result = await client.post(url, { form: params, body });
```
[the remark this answers is not in the document]
Comment (PradeepKumar100791) [reply lib/index.js:105]: we have done this to upload file - we only send body in that case

File: lib/util.js (hunk 1 above)
Comment (PradeepKumar100791) [reply lib/util.js:310]: Done

**Comment by durlabhjain** [comment]
Rejected as most changes are now in the main branch. Let me know if we still need it.

**Pull request closed without merging**